src/psyclone/tests/psyGen_test.py

```diff
@@ -2668,17 +2692,609 @@ def test_binaryoperation_can_be_printed():
+        sym.is_input = 3
+    assert "Symbol 'is_input' attribute must be a boolean." in \
+           str(error.value)
+
+    sym = Symbol('a', 'real', [], 'local')
+    with pytest.raises(ValueError) as error:
+        sym.is_input = True
+    assert ("Symbol with 'local' scope can not have 'is_input' attribute"
+            " set to True.") in str(error.value)
+
+
+def test_symbol_is_output_setter():
+    '''Test that a Symbol is_output can be set if given a new valid
+    value, otherwise it raises a relevant exception'''
+    sym = Symbol('a', 'real', [], 'global_argument', False, False)
+    sym.is_output = True
+    assert sym.is_output is True
+
+    with pytest.raises(TypeError) as error:
+        sym.is_output = 3
+    assert "Symbol 'is_output' attribute must be a boolean." in \
+           str(error.value)
+
+    sym = Symbol('a', 'real', [], 'local')
+    with pytest.raises(ValueError) as error:
+        sym.is_output = True
+    assert ("Symbol with 'local' scope can not have 'is_output' attribute"
+            " set to True.") in str(error.value)
+
+
+def test_symbol_can_be_printed():
+    '''Test that a Symbol instance can always be printed. (i.e. is
+    initialised fully)'''
+    symbol = Symbol("sname", "real")
+    assert "sname<real, [], local>" in str(symbol)
+
+
+# Test SymbolTable Class
+
+def test_symboltable_declare():
+    '''Test that the declare method inserts new symbols in the symbol
+    table, but raises appropiate errors when provied with wrong parameters
+    or duplicate declarations.'''
+    sym_table = SymbolTable()
+
+    # Declare a symbol
+    sym_table.declare("var1", "real", [5, 1], "global_argument", True, True)
+    assert sym_table._symbols["var1"].name == "var1"
+    assert sym_table._symbols["var1"].datatype == "real"
+    assert sym_table._symbols["var1"].shape == [5, 1]
+    assert sym_table._symbols["var1"].scope == "global_argument"
+    assert sym_table._symbols["var1"].is_input is True
+    assert sym_table._symbols["var1"].is_output is True
+
+    # Declare a duplicate name symbol
+    with pytest.raises(KeyError) as error:
+        sym_table.declare("var1", "real")
+    assert ("Symbol table already contains a symbol with name "
+            "'var1'.") in str(error.value)
+
+
+def test_symboltable_lookup():
+    '''Test that the lookup method retrives symbols from the symbol table
+    if the name exists, otherwise it raises and error.'''
+    sym_table = SymbolTable()
+    sym_table.declare("var1", "real", [None, None])
+    sym_table.declare("var2", "integer", [])
+    sym_table.declare("var3", "real", [])
+
+    assert isinstance(sym_table.lookup("var1"), Symbol)
+    assert sym_table.lookup("var1").name == "var1"
+    assert isinstance(sym_table.lookup("var2"), Symbol)
+    assert sym_table.lookup("var2").name == "var2"
+    assert isinstance(sym_table.lookup("var3"), Symbol)
+    assert sym_table.lookup("var3").name == "var3"
+
+    with pytest.raises(KeyError) as error:
```

Comment:
pylint points out that you don't use `error`

Reply:
Fixed